std/python/_std/sys/thread/Thread.hx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C)2005-2019 Haxe Foundation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package sys.thread;
+
+class Thread {
+	var nativeThread: NativeThread;
+	var messages: Deque<Dynamic>;
+
+	static var threads = new haxe.ds.ObjectMap<NativeThread, Thread>();
+	static var threadsMutex: Mutex = new Mutex();
+	static var mainThread: Thread;
+
+	private function new(t:NativeThread) {
+		nativeThread = t;
+		messages = new Deque<Dynamic>();
+	}
+
+	public function sendMessage(msg:Dynamic):Void {
+		messages.add(msg);
+	}
+
+	public static function current():Thread {
+		threadsMutex.acquire();
+		var ct = PyThreadingAPI.current_thread();
+		if (ct == PyThreadingAPI.main_thread()) {
+			if (mainThread == null) mainThread = new Thread(ct);
+			return mainThread;
```

Comment:
The mutex should be released before this return.

This also makes me thing no test currently calls `current()` on the main thread.  Since that's a common special case, it's probably worth adding something to cover it.

Reply:
I think that DequeBrackets calls current() on the main thread indirectly through readMessage(), but only once, which would explain why this bug wasn't caught. Maybe I could add another call to the DequeBrackets test?

Reply:
Weird. I would have expected that adding this line:

```
switch (Thread.readMessage(true)) {
	case "done":
	case s:
		Assert.fail("Unexpected message: " + s);
}
Thread.current();                      // THIS LINE IS NEW
```

would cause the test to hang, because the mutex wouldn't be released after readMessage() is called. But it still works for me.